File: 프로그래머스/LV1/최소직사각형.py
def solution(sizes):
    answer = 0
    for i in sizes:
        if i[0] < i[1]:
            i[0], i[1] = i[1], i[0]
    sizes.sort()
    for i in sizes:
        if i[1] > answer:
            answer = i[1]
    return answer * sizes[-1][0]

File: 프로그래머스/javascript/LV1/최소직사각형.js
function solution(sizes) {
  let x = 0;
  let y = 0;
  const changedSizes = sizes.map((e) => (e[0] < e[1] ? [e[1], e[0]] : e));
  for (let i = 0; i < changedSizes.length; i++) {
    if (changedSizes[i][0] > x) {
      x = changedSizes[i][0];
    }
    if (changedSizes[i][1] > y) {
      y = changedSizes[i][1];
    }
  }

  return x * y;
}
